File: python/examples/transform/wavelet.py
import sys
import numpy
from metric.transform import dwt, idwt, wmaxlev


def main():
    x = numpy.float_([0.1, 0.2, 0.3, 0.2, 0.5])
    a, b = dwt(x, 5)
    print(list(a))
    print(list(b))

    a = numpy.float_([0.1, 0.2, 0.3, 0.2, 0.5])
    b = numpy.float_([0.1, 0.11, 0.7, 0.2, 0.0])
    print(list(idwt(a, b, 5, 5)))

    print(wmaxlev(5, 5))


sys.exit(main())

File: python/src/transform/wavelet.cpp
#include "modules/transform/wavelet.hpp"

#include <pybind11/pybind11.h>
#include <pybind11/stl.h>
#include <pybind11/numpy.h>
#include <vector>

namespace py = pybind11;

void export_metric_wavelet(py::module& m) {
    using T = double;
    using Container = std::vector<T>;
    m.def("dwt", &wavelet::dwt<Container>,
        py::arg("x"),
        py::arg("wavelet_type")
    );
    m.def("idwt", &wavelet::idwt<Container>);
    m.def("wmaxlev", &wavelet::wmaxlev);
}
